addons/assignGear/functions/fnc_assignGearMan.sqf

```diff
@@ -13,27 +13,99 @@
  *
  * Public: Yes
  */
-
 #include "script_component.hpp"
 
-TRACE_1("params",_this);
-
 params ["_unit"];
-TRACE_2("",_unit, local _unit);
+TRACE_3("assignGearMan",_unit, local _unit, typeOf _unit);
 
 BEGIN_COUNTER(assignGearMan);
 
-private _faction = toLower faction _unit;
-private _unitClassname = [typeOf _unit] call FUNC(cleanPrefix);
+private _faction = faction _unit;
+private _typeOf = typeOf _unit;
+private _unitClassname = [_typeOf] call FUNC(cleanPrefix);
 private _loadout = _unit getVariable ["F_Gear", _unitClassname]; //Check variable f_gear, otherwise default to typeof
 private _path = missionConfigFile >> "CfgLoadouts" >> _faction >> _loadout;
 
 if ((!isClass(_path)) && GVAR(useFallback)) then {
     _path = missionConfigFile >> "CfgLoadouts" >> _faction >> "fallback";
 };
 
+// Temp? BWC for older missions
+if ((!isClass _path) && {(_typeOf select [0,7]) == "potato_"}) then {
+    private _converstionType = 0;
+    if (_faction =="potato_w" && {isClass (missionConfigFile >> "CfgLoadouts" >> "blu_f")}) then {
+        _faction ="blu_f";
+        _converstionType = 1;
+    };
+    if (_faction =="potato_i" && {isClass (missionConfigFile >> "CfgLoadouts" >> "ind_f")}) then {
+        _faction ="ind_f";
+        _converstionType = 1;
+    };
+    if (_faction =="potato_e" && {isClass (missionConfigFile >> "CfgLoadouts" >> "opf_f")}) then {
+        _faction ="opf_f";
+        _converstionType = 1;
+    };
+    if (_faction =="potato_e" && {isClass (missionConfigFile >> "CfgLoadouts" >> "potato_msv")}) then {
+        _faction ="potato_msv";
+        _converstionType = 2;
+    };
+    if (_converstionType == 0) exitWith { INFO_1("No BWC Factions %1",_typeOf); };
+    private _unitRole = toLower _typeOf select [9]; // simple because potato_x_ is const length
+    switch (_converstionType) do {
+        case (1): {
+            _loadout = switch (_unitRole) do {
+                case"rifleman": {"Soldier_F"};
+                case"ftl": {"Soldier_TL_F"};
+                case"coy";
+                case"xo";
+                case"sl": {"Soldier_SL_F"};
+                case"coy": {"officer_F"};
+                case"ar": {"Soldier_AR_F"};
+                case"aar": {"Soldier_AAR_F"};
+                case"lat": {"Soldier_LAT_F"};
+                case"cm";
```

Comment:
Not quite sure how this works to be honest, but there are platoon medic units as well, would you need to define them here awsell?